frontend/src/components/Form.js

```diff
@@ -0,0 +1,171 @@
+
+// to either LOGIN or REGISTER as a new user
+export const Form = () => {
+  const dispatch = useDispatch()
+  const accessToken = useSelector((store) => store.user.login.accessToken)
+  const [email, setEmail] = useState('')
+  const [password, setPassword] = useState('')
+  const [confirmedPassword, setConfirmedPassword] = useState('')
+  const [name, setName] = useState('')
+  const [isNewUser, setIsNewUser] = useState(false)
+  const [loginFailed, setLoginFailed] = useState(false)
+
+  const handleLoginSuccess = (loginResponse) => {
+    dispatch(user.actions.setAccessToken({ accessToken: loginResponse.accessToken }))
+    dispatch(user.actions.setUserId({ userId: loginResponse.userId }))
+    dispatch(user.actions.setStatusMessage({ statusMessage: 'Login Success' }))
+    setEmail('')
+    setName('')
+    setPassword('')
+    setConfirmedPassword('')
+    setIsNewUser(false)
+    setLoginFailed(false)
+  }
+
+  const handleLoginFail = (loginError) => {
+    dispatch(user.actions.setAccessToken({ accessToken: null }))
+    dispatch(user.actions.setStatusMessage({ statusMessage: loginError }))
+    setLoginFailed(true)
+  }
+
+  const handleLogin = (event) => {
+    event.preventDefault()
+
+    fetch(LOGIN_URL, {
+      method: 'POST',
+      body: JSON.stringify({ email, password }),
+      headers: { 'Content-Type': 'application/json' }
+    })
+      .then((res) => {
+        if (!res.ok) {
+          throw 'Login failed'
+        }
+        return res.json()
+      })
+      .then((json) => handleLoginSuccess(json))
+      .catch((err) => handleLoginFail(err))
+  }
+
+  const handleSignup = (event) => {
+    event.preventDefault()
+
+    fetch(SIGNUP_URL, {
+      method: 'POST',
+      body: JSON.stringify({ name, email, password }),
+      headers: { 'Content-Type': 'application/json' }
+    })
+      .then((res) => {
+        if (!res.ok) {
+          throw 'Signup failed'
+        }
+        return res.json()
+      })
+      .then((json) => handleLoginSuccess(json))
+      .catch((err) => handleLoginFail(err))
+  }
+
+  const handleNewUser = (event) => {
+    event.preventDefault()
+    setIsNewUser(true)
+    setEmail('')
+    setPassword('')
+  }
+  if (accessToken) {
+    return <Secrets />
+  }
+  if (!isNewUser) {
+    return (
+      <>
+        <FormStyle onSubmit={handleLogin}>
+          <Title>Login</Title>
+          <Input
+            type="email"
+            value={email}
+            placeholder="email"
+            onChange={(event) => setEmail(event.target.value)}
+            required />
+          <Input
+            type="password"
+            value={password}
+            placeholder="password"
+            onChange={(event) => setPassword(event.target.value)}
+            required />
+          <Button
+            type="submit">
+            Login
+          </Button>
+          {loginFailed && <Paragraph>Wrong email or password. Please try again.</Paragraph>}
+          <Button
+            type="submit"
```

Comment:
here you start to use double quotation marks " " instead of single ' '. Remember to be consistent :) It's the same for the sign up form

Reply:
We have double quotes in the form since eslint complains when we use single quotes. I wonder if this makes a difference though? Hmm!